x/finality/types/power_table.go

```diff
@@ -62,6 +62,22 @@ func (dc *VotingPowerDistCache) FindNewInactiveFinalityProviders(prevDc *VotingP
 		}
 	}
 
+	// Also check for finality providers that were active before but now
+	// are not present in the current cache due to full unbonding.
+	currFps := make(map[string]struct{})
+	for _, fp := range dc.FinalityProviders {
+		currFps[fp.BtcPk.MarshalHex()] = struct{}{}
+	}
+	prevActivePfs := prevDc.GetActiveFinalityProviderSet()
```

Comment:
Note here: at this point in event processing we are after the `k.RecordVotingPowerAndCache(ctx, newDc)` function 
which means both `dc` and `prevDc` contains only active finality providers.  

Does it mean whole begining of the function i.e:
```go
inactiveFps := dc.GetInactiveFinalityProviderSet()
prevInactiveFps := prevDc.GetInactiveFinalityProviderSet()
newInactiveFps := make([]*FinalityProviderDistInfo, 0)
for pk, fp := range inactiveFps {
  _, exists := prevInactiveFps[pk]
  if !exists {
	  newInactiveFps = append(newInactiveFps, fp)
  }
}
```

is essentially one big no-op that we can delete ? (`GetInactiveFinalityProviderSet` will always return empty set for both dcs)

Reply:
scratch that, I misunderstood something there 👍